src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -601,34 +581,62 @@ private void WriteIndexerBuildCommand(CodeIndexer indexer, CodeMethod codeElemen
 
         if (indexer.ReturnType.AllTypes.First().TypeDefinition is not CodeClass td) return;
 
-        AddCommandBuilderContainerInitialization(parent, targetClass, writer, prefix: $"var {BuilderInstanceName} = ", pathParameters: codeElement.Parameters.Where(x => x.IsOfKind(CodeParameterKind.Path)));
-        writer.WriteLine("var commands = new List<Command>();");
-
         var parentMethodNames = parent.Methods
             .Where(m => m.IsOfKind(CodeMethodKind.CommandBuilder) && !string.IsNullOrWhiteSpace(m.SimpleName))
             .Select(m => m.SimpleName)
             .ToHashSet(StringComparer.OrdinalIgnoreCase);
-            
+
         var builderMethods = td.Methods
             // If a method with the same name exists in the indexer's parent class, filter it.
-            .Where(m => m.IsOfKind(CodeMethodKind.CommandBuilder) && !parentMethodNames.Contains(m.SimpleName))
-            // Show executable commands first
-            .OrderBy(static m => m.OriginalMethod?.HttpMethod is null) ??
+            .Where(m => m.IsOfKind(CodeMethodKind.CommandBuilder) && !parentMethodNames.Contains(m.SimpleName)) ??
             Enumerable.Empty<CodeMethod>();
+        if (!builderMethods.Any())
+        {
+            writer.WriteLine($"return new(new(0), new(0));");
+            return;
+        }
+        bool hasExecutable = false;
+        bool hasNonExecutable = false;
+        var sb = new StringBuilder();

```

Comment:
avoid using intermediate string builders, it duplicates some of the concerns (indents, line breaks) and make the code harder to read.

Reply:
Removed string builder by using Any and checking early.